src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -81,6 +81,10 @@
    [this jruby-instance request-options]
    (.compileCatalog jruby-instance request-options))
 
+  (compile
```

Comment:
this conflicts with a clojure var, I think I'll rename it to `compile-ast`, you're working on the Ruby side, right @Magisus ? Do you have thoughts on the protocol name? Should I implement the java interface as part of this work too?

Reply:
renamed to `compile-ast` and implemented interface per convo in slack.